src/hdmf_zarr/backend.py

```diff
@@ -157,10 +157,19 @@ def object_codec_class(self):
     def open(self):
         """Open the Zarr file"""
         if self.__file is None:
-            self.__file = zarr.open(store=self.path,
-                                    mode=self.__mode,
-                                    synchronizer=self.__synchronizer,
-                                    storage_options=self.__storage_options)
+            # Within zarr, open_consolidated only allows the mode to be 'r' or 'r+'.
+            # As a result, when in 'w' mode, the file will not use consolidated metadata.
+            if self.__mode == 'w':
```

Comment:
What about modes `'a'` and  `'w-'`? Since open_cosolidated only allows for 'r' and 'r+' it may be easier to check for `if self.__mode not in ['r', 'r+']`